addons/website_sale_wallet/models/sale_order.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+from odoo.tools import float_compare
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    # Relational fields
+    wallet_product_id = fields.Many2one('product.product', store=False)
```

Comment:
Need to be a compute in this case.
Not set by another public method add_update_wallet_transaction that call a GET function _get_wallet_product that create smthing...
don't use default_code, it can be updated and have no many sense

What is the difference between wallet_product_id and  order_line_wallet_id.product_id ???

Reply:
they are the same ( we use wallet_product_id to create the order_line)

Reply:
What will I be using instead of default_code?